### ISSUE TITLE
False positive for E0602?

### ISSUE THREAD

**Issue by KommuSoft** [PyCQA/pylint#10562]
### Bug description

If I run pylint on the following:

```python
item = (1,4,2,5)
for item in item:
    print(item)
```

(I agree this is *very bad* practice, I encountered this while cleaning up an ugly codebase)

I get the following error:

```
file.py:2:12: E0602: Undefined variable 'item' (undefined-variable)
```

But this makes not much sense, since the variable is defined. Likely pylint reasons that since `for` assign a value to `item`, at the time the `item` is read as source of items, `item` is not defined, but it is.

Now it is definitely a good idea to generate a warning for this, but the variable is defined, so I think this is a false positive for E0602?

### Command used

```shell
pylint file.py
```

### Pylint output

```python
************* Module file
file.py:1:0: C0114: Missing module docstring (missing-module-docstring)
file.py:2:12: E0602: Undefined variable 'item' (undefined-variable)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

Not mentioning E0602.

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.2
Python 3.12.3 (main, Aug 14 2025, 17:47:21) [GCC 13.3.0]
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue, I can reproduce and I agree.

**Comment by Pierre-Sassoulas** [comment]
Will be released in 3.3.9 soonish :)
